### PR TITLE
improvement: remove Object3D as superclass from Cognite* models and refactor SceneHandler

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
# Description

This is done as a step towards better flexibility in visual tests using `StreamingVisualTestFixture`. Also fits better with our philosophy of prefering `CognitePointCloudModel` and `Cognite3dModel` for external use only.

# Checklist:

Here is a checklist that should completed before merging this given feature. 
Any shortcomings from the items below should be explained and detailed within the contents of this PR.

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

**Comment by christjt** [comment]
I think this makes a lot of sense, but lets defer it to 4.0. Since this sort of requires us to remove the Cognite3DModel/PointCloud being an Object3D.

**Comment by haakonflatval-cognite** [comment]
Will delay further development on this until 4.0

**Comment by haakonflatval-cognite** [comment]
Solves https://cognitedata.atlassian.net/browse/REV-501

File: viewer/packages/cad-model/visual-tests/DebugLoadedSectorsTool.ts [hunk 1]
```diff
@@ -6,29 +6,28 @@ import * as THREE from 'three';
 //TODO: Refactor such that these are properly imported (tools should not be dependent on cad-parser)
 import { LevelOfDetail, SectorNode } from '../../cad-parsers/';
 
-import { Cognite3DViewerToolBase } from './Cognite3DViewerToolBase';
+// import { Cognite3DViewerToolBase } from './Cognite3DViewerToolBase';
```
Comment (pramodcog) [inline viewer/packages/cad-model/visual-tests/DebugLoadedSectorsTool.ts:9]:
```suggestion
```

File: viewer/packages/cad-model/visual-tests/DebugLoadedSectorsTool.ts [hunk 2]
```diff
@@ -6,29 +6,28 @@ import * as THREE from 'three';
...
 import { assertNever } from '@reveal/utilities';
-import { Cognite3DModel, Cognite3DViewer } from '@reveal/api';
+// import { Cognite3DModel } from '@reveal/api';
```
Comment (pramodcog) [inline viewer/packages/cad-model/visual-tests/DebugLoadedSectorsTool.ts:11]:
```suggestion
```

**Review by pramodcog — APPROVED** [review]
Looks good!

**Pull request merged**